src/ui/egg-gacha-ui-handler.ts

```diff
@@ -684,150 +694,165 @@ export default class EggGachaUiHandler extends MessageUiHandler {
-                  error = true;
-                  this.showError(i18next.t("egg:notEnoughVouchers"));
-                } else if (globalScene.gameData.eggs.length < 75 || Overrides.UNLIMITED_EGG_COUNT_OVERRIDE) {
-                  if (!Overrides.EGG_FREE_GACHA_PULLS_OVERRIDE) {
-                    this.consumeVouchers(VoucherType.GOLDEN, 1);
-                  }
-                  this.pull(25);
-                  success = true;
-                } else {
-                  error = true;
-                  this.showError(i18next.t("egg:tooManyEggs"));
-                }
-                break;
-              case 5:
-                ui.revertMode();
-                success = true;
-                break;
-            }
-            break;
-          case Button.CANCEL:
-            this.getUi().revertMode();
-            success = true;
-            break;
-          case Button.UP:
-            if (this.cursor) {
-              success = this.setCursor(this.cursor - 1);
-            }
-            break;
-          case Button.DOWN:
-            if (this.cursor < 5) {
-              success = this.setCursor(this.cursor + 1);
-            }
-            break;
-          case Button.LEFT:
-            if (this.gachaCursor) {
-              success = this.setGachaCursor(this.gachaCursor - 1);
-            }
-            break;
-          case Button.RIGHT:
-            if (this.gachaCursor < getEnumKeys(GachaType).length - 1) {
-              success = this.setGachaCursor(this.gachaCursor + 1);
-            }
-            break;
+        break;
+      case Button.DOWN:
+        if (this.cursor < 5) {
+          success = this.setCursor(this.cursor + 1);
         }
-      }
+        break;
+      case Button.LEFT:
+        if (this.gachaCursor) {
+          success = this.setGachaCursor(this.gachaCursor - 1);
+        }
+        break;
+      case Button.RIGHT:
+        if (this.gachaCursor < Object.keys(GachaType).length - 1) {
+          success = this.setGachaCursor(this.gachaCursor + 1);
+        }
+        break;
     }
 
-    if (success) {
-      ui.playSelect();
-    } else if (error) {
-      ui.playError();
+    // Return undefined here because we do not play error sound in case of
+    return success || undefined;
+  }
+
+  /**
+   * Handles an input event that occurs while the egg gacha summary is visible
+   * @param button - The button that was pressed
+   * @returns `true` if an input event occurred and the select sound should be played, otherwise `undefined`
+   */
+  private processSummaryInput(button: Button): true | undefined {
+    if (this.summaryFinished && (button === Button.ACTION || button === Button.CANCEL)) {
+      this.hideSummary();
+      return true;
     }
+  }
 
-    return success || error;
+  /**
+   *
+   * @param button - The button that was pressed
+   * @returns - Whether an input event occured.
+   */
+  processInput(button: Button): boolean {
+    let success: boolean | undefined = undefined;
```

Comment:
```suggestion
    let success: boolean | undefined;
```
I know Biome complains if you do `let x: y | undefined = undefined;`.